test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -256,6 +263,60 @@ protected Layout getManagementTestLayout()
         return l;
     }
 
+    /**
+     * Scenario with 1 node: SERVERS.PORT_0
+     * The node is setup, bootstrapped and then requested for a
+     * heartbeat. This is responded with the nodeMetrics which is
+     * asserted with expected values.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void checkHeartbeat()
+            throws Exception {
+        addServer(SERVERS.PORT_0);
+
+        Layout l = new TestLayoutBuilder()
+                .setEpoch(1L)
+                .addLayoutServer(SERVERS.PORT_0)
+                .addSequencer(SERVERS.PORT_0)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(SERVERS.PORT_0)
+                .addToSegment()
+                .addToLayout()
+                .build();
+        bootstrapAllServers(l);
+
+
+        CorfuRuntime corfuRuntime = getRuntime(l).connect();
+
+        // Set aggressive timeouts.
+        setAggressiveTimeouts(l, corfuRuntime,
+                getManagementServer(SERVERS.PORT_0).getManagementAgent().getCorfuRuntime());
+
+        NodeView nodeView = null;
+
+        ServerMetrics serverMetrics =
+                new ServerMetrics(NodeLocator.parseString(SERVERS.ENDPOINT_0),
+                        new SequencerMetrics(SequencerStatus.UNKNOWN));
+
+        // Send heartbeat requests and wait until we get a valid response.
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_LOW; i++) {
+
+            nodeView = corfuRuntime.getLayoutView().getRuntimeLayout()
+                    .getManagementClient(SERVERS.ENDPOINT_0).sendHeartbeatRequest().get();
+
+            if (!nodeView.getEndpoint().toString().isEmpty()) {
+                break;
+            }
+            Thread.sleep(PARAMETERS.TIMEOUT_VERY_SHORT.toMillis());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)